extensions/emmet/src/util.ts

```diff
@@ -380,30 +380,34 @@ export const allowedMimeTypesInScriptTag = ['text/html', 'text/plain', 'text/x-t
  * If position is inside a script tag of type template, then it will be parsed to find the inner HTML node as well
  */
 export function getHtmlFlatNode(documentText: string, root: FlatNode | undefined, offset: number, includeNodeBoundary: boolean): HtmlFlatNode | undefined {
-	const currentNode: HtmlFlatNode | undefined = <HtmlFlatNode | undefined>getFlatNode(root, offset, includeNodeBoundary);
+	let currentNode: HtmlFlatNode | undefined = <HtmlFlatNode | undefined>getFlatNode(root, offset, includeNodeBoundary);
 	if (!currentNode) { return; }
 
-	const isTemplateScript = currentNode.name === 'script' &&
-		(currentNode.attributes &&
-			currentNode.attributes.some(x => x.name.toString() === 'type'
+	// If the currentNode is a script one, first set up its subtree and then find HTML node.
+	if (currentNode.name === 'script') {
+		setUpScriptNodeSubtree(documentText, currentNode);
+		currentNode = <HtmlFlatNode | undefined>getFlatNode(root, offset, includeNodeBoundary);
```

Comment:
Do we have to re-search from the root in this case?

Reply:
@vinku bump

Reply:
I will try if we can restrict it today.

Reply:
ok, searching inside currentNode (i.e the script node, if that returns null, then the script-node is the node we need).